core/store/src/db.rs

```diff
@@ -79,6 +80,119 @@ impl DBTransaction {
 
 pub type DBIterator<'a> = Box<dyn Iterator<Item = io::Result<(Box<[u8]>, Box<[u8]>)>> + 'a>;
 
+/// Data returned from the database.
+///
+/// Abstraction layer for data returned by [`Database::get_raw_bytes`] and
+/// [`Database::get_with_rc_stripped`] methods.  Operating on the value as
+/// a slice is free while converting it to a vector on an arc may requires an
+/// allocation and memory copy.
+pub struct DBBytes<'a>(DBBytesInner<'a>);
```

Comment:
Something which annoys me a tiny bit is that, according to Rust naming guidelines, stuff should be `DbTransaction`, `DbBytes` and such. Obviously not a good idea to change this here